OneCommunityGlobal/HighestGoodNetworkApp
# Yufei - make intangible time numbers having a mouseover explanation editable by the owner

# Description
<img width="738" alt="Screen Shot 2023-07-21 at 16 31 39" src="https://github.com/OneCommunityGlobal/HGNRest/assets/111897624/458c58e4-ca45-4bf8-baac-758f123ae4d4">

## Related PRS (if any):
This frontend PR is related to the [#454](https://github.com/OneCommunityGlobal/HGNRest/pull/454) backend PR.

## Main changes explained:
- Create mouseover text of total time component and add it into leaderboard component
- Create mouseover text action, reducer, etc.


## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log in as an owner user and verify that the mouseover text appears when you move your mouse over both 'Total Time' and the numbers on the Leaderboard. Some numbers are blue (indicating intangible hours logged), and some numbers are black. Mouseover text will appear on both types of numbers.
4. Only the owner can see the edit button next to 'Total Time'. Click on it and enter the new text to update the mouseover text. Please note that it may take some time to respond, as shown in my screen recording.
5. Wait a few seconds for a green pop-up box to appear, showing the message "Mouseover Text updated!" Then you will see the newly updated mouseover text when you move your mouse over both 'Total Time' and the numbers. If you don't immediately see the changes, just wait a few more seconds, as it may take some time for the updates to respond.
6. Hard refresh the page to see the change is globally. The same thing is if you don't immediately see the changes, just wait a few more seconds, as it may take some time for the updates to respond.
7. login as any other role to verify the mouseoverText updates globally and they do not see the edit button.

## Screenshots or videos of changes:
Log in as an owner:
<img width="648" alt="Screen Shot 2023-07-21 at 19 43 12" src="https://github.com/OneCommunityGlobal/HGNRest/assets/111897624/15d7b4a6-8ca6-4081-be24-0a88f8bca382">


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111897624/4ceff3aa-ae86-4ba2-8dfa-096808022cb5


Log in as any other user except owner:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111897624/46e3df42-b839-4469-bd34-afc8113ecf54









## Note:
Sometimes, you may need to wait for the page to respond and display the mouseover text. I encountered a similar situation as shown in my screenshot recording. When you move your mouse to another place where there should be mouseover text, you may experience the same situation. Just be patient and wait.
